frontend/src/routes/_protected/treecluster/index.tsx

```diff
@@ -1,13 +1,45 @@
+import { useState, useEffect } from 'react';
 import TreeclusterCard from "@/components/general/cards/TreeclusterCard";
+import Dialog from "@/components/general/filter/Dialog";
 import { treeclusterDemoData } from "@/data/treecluser";
-import { createFileRoute } from "@tanstack/react-router";
+import { createFileRoute, useLoaderData } from '@tanstack/react-router';
+import { z } from 'zod';
 
-export const Route = createFileRoute("/_protected/treecluster/")({
+const treeclusterFilterSchema = z.object({
+  status: z.array(z.string()).optional(),
+  region: z.array(z.string()).optional(),
+});
+
+export const Route = createFileRoute('/_protected/treecluster/')({
   component: Treecluster,
+  validateSearch: treeclusterFilterSchema,
+
+  loaderDeps: ({ search: { status, region } }) => ({
+    status: status || [],
+    region: region || [],
+  }),
+
+  loader: ({ deps: { status, region } }) => {
+    return { status, region };
+  },
 });
 
 function Treecluster() {
   const clusters = treeclusterDemoData();
+  const search = useLoaderData({ from: '/_protected/treecluster/' });
+
+  const [statusTags, setStatusTags] = useState<string[]>(search.status);
+  const [regionTags, setRegionTags] = useState<string[]>(search.region);
+
+  useEffect(() => {
+    if (search.status) setStatusTags(search.status);
+    if (search.region) setRegionTags(search.region);
+  }, [search.status, search.region]);
```

Comment:
And because we have parsed it in the router loader, we can access the data with the `useLoaderData` hook.